src/main.cpp

```diff
@@ -1,17 +1,183 @@
+ * - reach_cruise_height: Fly drone to the crusing height.
+ * - fly_to_waypoint: Executes the mode to fly to the waypoint.
+ * - reach_target_height: Fly drone to the target height.
+ * - post_wait_time: Wait a predefined time.
+ *
+ * If the node state is unknown, an error message is logged and a job_finished
+ * error message is sent.
+ */
+void WaypointNode::event_loop() {
+    if (!this->get_active()) return;
+
+    switch (get_node_state()) {
+        case init:
+            mode_init();
+            break;
+        case pre_wait_time:
+            break;  // Do nothing
+        case reach_cruise_height:
+            mode_reach_cruise_height();
+            break;
+        case fly_to_waypoint:
+            mode_fly_to_waypoint();
+            break;
+        case reach_target_height:
+            mode_reach_target_height();
+            break;
+        case post_wait_time:
+            break;  // Do nothing
+        default:
+            RCLCPP_ERROR(this->get_logger(),
+                         "WaypointNode::event_loop: Unknown mission_state: %d",
+                         get_node_state());
+            this->job_finished(
+                "WaypointNode::event_loop: Unknown mission_state");
+            reset_node();
+    }
+}
+
+/**
+ * @brief Resets the node for the next command.
+ *
+ * This function resets the state of the `WaypointNode` object for the next
+ * command. It sets the node state to `init`, sets the `state_first_loop` flag
+ * to `true`, resets the `mission_progress` to 0, and clears the `cmd` object.
+ */
+void WaypointNode::reset_node() {
+    set_node_state(init);
+    state_first_loop = true;
+    mission_progress = 0.0;
+    cmd = Command();
+}
+
+/**
+ * @brief Sets the node state to a new value.
+ *
+ * This function sets the node state to the specified new value. If the new
+ * value is different from the current node state, the `state_first_loop` flag
+ * is set to true. After updating the node state, a debug message is logged.
+ * Additionally, `mission_progress` is reset to 0.
+ *
+ * @param new_mission_state The new mission state to set.
+ */
+void WaypointNode::set_node_state(NodeState_t new_state) {
+    if (node_state != new_state) {
+        state_first_loop = true;
+    }
+
+    mission_progress = 0.0;
+    node_state = new_state;
+    RCLCPP_DEBUG(this->get_logger(),
+                 "WaypointNode::set_node_state: Set node state to %d",
+                 node_state);
+}
+
+/**
+ * @brief Checks if the command and position values are set.
+ *
+ * This function is used to check if the command and position values are set
+ * before executing a command. If either the command or position values are not
+ * set, the function logs an error message and returns false.
+ *
+ * @param function_name The name of the calling function.
+ * @return True if both the command and position values are set, false
+ * otherwise.
+ */
+bool WaypointNode::check_cmd(const char *function_name) {
+    if (!cmd.values_set) {
```

Comment:
You could add for both cases log messages that you are resetting the node as a consequence, but that is up to you.

Reply:
Changed in https://github.com/THI-Drone/waypoint_package/pull/3/commits/5a49068ac86cdab12bbed671d77177ca7fee6d9a